OSDOCS-19946: Reorganizing securing routes modules (#112654)
openshift/openshift-docs: networking/ingress_load_balancing/routes/securing-routes.adoc

toc::[]

[role="_abstract"]

> To secure application traffic, you can configure routes to serve custom certificates to clients using edge, passthrough, or re-encrypt TLS termination, as well as manage externally provided certificates. Additionally, you can enforce strict security protocols by securing a route with HTTP strict transport security (HSTS).

Reviewer: It would be highly valuable to list the specific failure reasons that appear in `oc describe route` for this feature. Specifically, `ExternalCertificateGetFailed` usually indicates an RBAC/SAR issue, while `ExternalCertificateValidationFailed` often means the secret exists but is the wrong type (not `kubernetes.io/tls`). Documenting these strings directly helps advanced users debug without needing to consult the router logs.

Author: Thanks! This is great info to add, but it probably shouldn't go in the assembly. I've updated nw-ingress-route-secret-load-external-cert.adoc to include this info.

> To secure application traffic, you can configure routes to serve custom certificates to clients using edge, passthrough, or re-encrypt TLS termination, as well as manage externally provided certificates. Additionally, you can enforce strict security protocols by securing a route with HTTP strict transport security (HSTS).

Reviewer 2: ```suggestion
To secure application traffic, you can configure routes to serve custom certificates to clients by using edge, passthrough, or re-encrypt TLS termination, and manage externally provided certificates. Additionally, you can enforce strict security protocols by securing a route with HTTP strict transport security (HSTS).
```

Author: got it, thanks

//Creating an edge route with a custom certificate
include::modules/nw-ingress-creating-an-edge-route-with-a-custom-certificate.adoc[leveloffset=+1]

//Creating a reencrypt route with a custom certificate
include::modules/nw-ingress-creating-a-reencrypt-route-with-a-custom-certificate.adoc[leveloffset=+1]

//Creating a passthrough route
include::modules/nw-ingress-creating-a-passthrough-route.adoc[leveloffset=+1]

//Creating a route with externally managed certificates
include::modules/nw-ingress-route-secret-load-external-cert.adoc[leveloffset=+1]

//HTTP Strict Transport Security

> include::modules/nw-enabling-hsts.adoc[leveloffset=+1]

Reviewer: With the introduction of header manipulation in `IngressController` (4.22+), we should explicitly recommend the `haproxy.router.openshift.io/hsts_header` annotation as the primary method for HSTS. Manually injecting `Strict-Transport-Security` via the new general-purpose header API while the HSTS annotation is also present can lead to duplicate headers or non-deterministic behavior in the HAProxy configuration.

Author: Thanks! I put this info in nw-enabling-hsts.adoc

Reviewer: Based on looking at a 4.21 cluster and considering 4.22 changes: The `IngressController` API schema for `spec.httpHeaders.actions` now explicitly reserves and blocks the `Strict-Transport-Security` header. The HSTS annotation is not just the 'recommended' method—it is the **only supported method** for configuring this header at the Route level. We should ensure the documentation reflects that manual header injection via the new API will be rejected by the cluster.

Reviewer 3: No, this is not true.  Can we discuss directly outside of this review?
